core/arch/arm/kernel/link.mk

```diff
@@ -92,9 +93,37 @@ $(link-script-pp): $(link-script) $(link-script-extra-deps)
 	$(q)$(CPPcore) -Wp,-P,-MT,$@,-MD,$(link-script-dep) \
 		$(link-script-cppflags) $< > $@
 
+define update-buildcount
+	@$(cmd-echo-silent) '  UPD     $(1)'
+	$(q)if [ ! -f $(1) ]; then \
+		mkdir -p $(dir $(1)); \
+		echo 1 >$(1); \
+	else \
+		expr 0`cat $(1)` + 1 >$(1); \
+	fi
+endef
+
+define gen-version-o
```

Comment:
Is not it the generation of the .c file?  In that case, does it work because LD is first compiling the .c file in .o (but potentially with wrong options)?